src/main/java/frc/robot/subsystems/Turret/Turret.kt

```diff
@@ -25,8 +27,18 @@ object Turret : Subsystem {
 
     private val turretInputs = TurretInputs()
 
+    const val MAX_ROTATION_DEGREES = 90.0
     // factoring in rotation of drivetrain
     private var targetRotation: Rotation2d = Rotation2d()
+        set(value) {
+            var degrees = (value.degrees % 360)
+            // spin around to other side if possible
+            if (degrees.absoluteValue > 180) {
+                degrees = (degrees.absoluteValue - 180) * -degrees.sign
+            }
```

Comment:
consider just checking if target is in the correct interval, if it is go there, and if it isn't just coerce it in like you were doing.